test/e2e/chaos/basic.go

```diff
@@ -1726,65 +1637,126 @@ selector:
-				framework.ExpectEqual(allSlowConnection(), [][]int{{0, 1}, {0, 2}})
+				framework.ExpectEqual(allSlowConnection(c, networkPeers, ports), [][]int{{0, 1}, {0, 2}})
 
 				err = cli.Delete(ctx, networkDelay.DeepCopy())
 				framework.ExpectNoError(err, "delete network chaos error")
 				time.Sleep(5 * time.Second)
-				framework.ExpectEqual(allSlowConnection(), [][]int{{0, 2}})
+				framework.ExpectEqual(allSlowConnection(c, networkPeers, ports), [][]int{{0, 2}})
 				err = cli.Delete(ctx, evenNetworkDelay.DeepCopy())
 				framework.ExpectNoError(err, "delete network chaos error")
 				time.Sleep(5 * time.Second)
-				framework.ExpectEqual(len(allSlowConnection()), 0)
-
-				complicateNetem := &v1alpha1.NetworkChaos{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      "network-chaos-3",
-						Namespace: ns,
-					},
-					Spec: v1alpha1.NetworkChaosSpec{
-						Action: v1alpha1.DelayAction,
-						Selector: v1alpha1.SelectorSpec{
-							Namespaces:     []string{ns},
-							LabelSelectors: map[string]string{"app": "network-peer-0"},
+				framework.ExpectEqual(len(allSlowConnection(c, networkPeers, ports)), 0)
+
+				complicateNetem := makeNetworkDelayChaos(
+					ns, "network-chaos-3",
+					map[string]string{"app": "network-peer-0"},
+					v1alpha1.OnePodMode,
+					v1alpha1.TcParameter{
+						Delay: &v1alpha1.DelaySpec{
+							Latency:     "200ms",
+							Correlation: "25",
+							Jitter:      "0ms",
 						},
-						Mode: v1alpha1.OnePodMode,
-						TcParameter: v1alpha1.TcParameter{
-							Delay: &v1alpha1.DelaySpec{
-								Latency:     "200ms",
-								Correlation: "25",
-								Jitter:      "0ms",
-							},
-							Loss: &v1alpha1.LossSpec{
-								Loss:        "25",
-								Correlation: "25",
-							},
-							Duplicate: &v1alpha1.DuplicateSpec{
-								Duplicate:   "25",
-								Correlation: "25",
-							},
-							Corrupt: &v1alpha1.CorruptSpec{
-								Corrupt:     "25",
-								Correlation: "25",
-							},
+						Loss: &v1alpha1.LossSpec{
+							Loss:        "25",
+							Correlation: "25",
 						},
-						Duration: pointer.StringPtr("9m"),
-						Scheduler: &v1alpha1.SchedulerSpec{
-							Cron: "@every 10m",
+						Duplicate: &v1alpha1.DuplicateSpec{
+							Duplicate:   "25",
+							Correlation: "25",
+						},
+						Corrupt: &v1alpha1.CorruptSpec{
+							Corrupt:     "25",
+							Correlation: "25",
 						},
 					},
-				}
+					pointer.StringPtr("9m"),
+					&v1alpha1.SchedulerSpec{
+						Cron: "@every 10m",
+					},
+				)
+
 				klog.Infof("Injecting delay for 0")
 				err = cli.Create(ctx, complicateNetem.DeepCopy())
 				framework.ExpectNoError(err, "create network chaos error")
 				time.Sleep(5 * time.Second)
-				framework.ExpectEqual(allSlowConnection(), [][]int{{0, 1}, {0, 2}, {0, 3}})
+				framework.ExpectEqual(allSlowConnection(c, networkPeers, ports), [][]int{{0, 1}, {0, 2}, {0, 3}})
+
+				cancel()
+			})
+		})
+
+		ginkgo.Context("[NetworkPartitionWithNetem]", func() {
```

Comment:
I'm not sure whether this test case meets the requirements from issue #1048, since it want to test ipset rules works with iptables changes, and I have no idea about to add the corner case with potential conflicts for them. Any suggestions is welcome :)

Reply:
`Netem` related NetworkChaos adds `iptables` rules iff the `.Spec.Target` field is set.